libs/ledgerjs/packages/cryptoassets/src/tokens.ts

```diff
@@ -21,23 +21,38 @@ const tokensByTicker: Record<string, TokenCurrency> = {};
 const tokensByAddress: Record<string, TokenCurrency> = {};
 const tokensByCurrencyAddress: Record<string, TokenCurrency> = {};
 const tokenListHashes = new Set();
-addTokens(erc20tokens.map(convertERC20));
+
+addTokens(
+  erc20tokens
+    .filter(([parentCurrencyId]) => !!findCryptoCurrencyById(parentCurrencyId))
+    .map(convertERC20),
+);
```

Comment:
I think it was already raised by @lambertkevin [here](https://github.com/LedgerHQ/ledger-live/pull/5443#discussion_r1395590634), but traversing 2 times a penitentially large array could bring some performance concerns.

Here is a suggestion combining the `filter` and `map` operations into a single `reduce` operation to iterate over the array only once, thus optimizing the time complexity.

```suggestion
addTokens(
  erc20tokens.reduce((acc: TokenCurrency[], token: ERC20Token) => {
    const [parentCurrencyId] = token;
    const currency = findCryptoCurrencyById(parentCurrencyId);
    if (currency) {
      acc.push(convertERC20(token));
    }
    return acc;
  }, []),
);
```

Reply:
Alternatively, we could sanitize the token list right at the import step in `libs/ledgerjs/packages/cryptoassets/src/data/erc20.ts` and filter out tokens without a parent currency right there at the source.

If having a token without an associated parent currency available in the Live can cause issues, it could be safer to treat it as close as possible to the source.

Reply:
Indeed you can do that with only 1 loop if it was necessary, but like I was saying on Slack: My proposed change has a 0 cost/impact contrary to anything we'll do here. 
I initially proposed to simply change the `getCryptoCurrencyById` in `convertERC20` which would throw for a `findCryptoCurrencyById` with an an early return instead, which is already handled by `addTokens`.

Surprise suprise: `getCryptoCurrencyById` is litteraly calling `findCryptoCurrencyById` and throwing if the result is undefined. So exactly the same result, no unecessary changes here, and no changes of the `erc20.json` file in anyway.